stock_quantity_history_location/static/src/components/inventory_report/inventory_report.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template" xml:space="preserve">
+    <t t-inherit="stock.InventoryReport.Buttons" t-inherit-mode="extension" owl="1">
+        <button t-on-click="onClickApplyAll" position="before">
+            <t t-if="this.multi_location" name="has_multi_location">
+                <button
+                    id="inventory_date_location"
+                    t-if="
+                        props.resModel == 'stock.quant'
+                        and (
+                            !props.context.inventory_mode
+                            or props.context.inventory_report_mode
+                        )
+                        and !props.context.no_at_date
+                    "
+                    type="button"
+                    class="btn btn-primary o_button_at_date me-2"
+                    t-on-click="onClickInventoryAtDate"
+                >
+                    Inventory at Date &amp; Location
+                </button>
+            </t>
+            <t t-else="" name="has_not_multi_location" />
+        </button>
+        <t name="has_not_multi_location" position="inside">
+            <xpath expr="//button[contains(@t-if, 'stock.quant')]" position="move" />
```

Comment:
This selector is a title bit ambiguous, I would use
```suggestion
            <xpath expr="//button[hasclass('o_button_at_date')]" position="move" />
```

![image](https://github.com/OCA/stock-logistics-reporting/assets/35231827/de71fee2-dd9b-4f7c-a8bf-f07bb6385c0e)

Reply:
Wait, I'm not sure why you need another button if your button is exactly the same as the original button(if and click-method)

Reply:
Actually, hasclass('o_button_at_date') is more ambiguous because stock_account add another button with that class, actually using the model is more specific for every button. What do you think? (This module will work together with stock_account_quantity_history_location if stock_account_quantity_history_location is migrated).

Reply:
I agree with @rolandojduartem, current filter is preferred to using class.

Regarding this:

> Wait, I'm not sure why you need another button if your button is exactly the same as the original button(if and click-method)

Because the text is different and there's no way to change or hide current text.

Reply:
you can use a t-out like

<button t-out="'demo'">imanie</button>